src/scalar.cpp

```diff
@@ -233,22 +233,15 @@ uint64_t bn_rshb_low(uint64_t *c, const uint64_t *a, int size, int bits)
     return carry;
 }
 
-#ifdef __USE_GMP
-// This is how GMP could be used under the hood for modular reduction. However, benchmarks show zero difference
-// to the cpp implementation used in this library (borrowed from relic-toolkit). See #else case below.
-#include <gmp.h>
-void bn_divn_low(uint64_t *c, uint64_t *d, uint64_t *a, int sa, uint64_t *b, int sb)
-{
-	mpn_tdiv_qr(c, d, 0, a, sa, b, sb);
-}
-#else
 void bn_divn_low(uint64_t *c, uint64_t *d, uint64_t *a, int sa, uint64_t *b, int sb)
 {
     int norm, i, n, t, sd;
     uint64_t carry, t1[3], t2[3];
 
     // Normalize x and y so that the leading digit of y is bigger than 2^(RLC_DIG-1).
-    norm = (64 - __builtin_clzll(b[sb - 1])) % 64;
+    // std::countl_zero will return 64 if uint64_t(0) is passed in. 
+    // The %64 operation will handle this case.
+    norm = (64 - std::countl_zero(b[sb - 1])) % 64;
```

Comment:
So if `b[sb - 1]) == 0`, norm will be 0, and we'll go into this `if(norm < 64 - 1)` branch below and set norm to 63. Is this the correct thing to do?

Reply:
OK, answering my own question, I checked in [relic](https://github.com/relic-toolkit/relic/blob/9548d266c92423436e695032bf54d6d08cd33a99/src/arch/lzcnt.inc#L44) and indeed the `clz` function they use (at least the generic version) returns `64` when passed `0`, just like `std::countl_zero()`. So we should be good.